exportlib.php

```diff
@@ -84,47 +84,41 @@ public function create_file($courseid, $feedbackid) {
 }
 
 /**
- * @author Felix Di Lenarda
- * rankings CSV export class
+ * Export feedback data for an entire feedbacj.
+ *
+ * @package block
+ * @subpackage coursefeedback
+ * @copyright innoCampus, TU Berlin
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-class rankingexport {
+class ranking_exporter {
     protected $feedback = null;
     protected $questionid = null;
-    protected $seperator = ";";
-
-    private $content = "";
+    protected $csvexportwriter;
 
     public function __construct($feedback = null, $question = null) {
-        global $DB;
-
-        if ($fb = $DB->get_record("block_coursefeedback", array("id" => $feedback))) {
-            $this->feedback = $fb;
-            $this->questionid = $question;
-        } else {
-            print_error("feedbacknotfound", "error");
-            exit(0);
-        }
+        $this->csvexportwriter = new csv_export_writer();
```

Comment:
Der Konstruktor verwirft nun komplett die Argumente `$feedback` und `$question`. Die werden auch als Attribute  gar nicht mehr benötigt, weil ja alles an `create_file` übergeben wird und dort passiert. Da stellt sich mir die Frage, ob es den Konstruktor überhaupt braucht. _Effektiv_ läuft es ja eigentlich darauf hinaus, dass `create_file` als statische Methode unmittelbar nach der Instantiierung eines `ranking_exporter` aufgerufen wird. Also das `$csvexportwriter` Attribut wird eigentlich auch nie wiederverwendet.

Ich weiß nicht, ob es gut wäre, da einfach komplett auf die Klasse zu verzichten, wenn es effektiv nur eine Funktion ist. Wenn nicht, dann sollte man zumindest die beiden überflüssigen Attribute und Konstruktor-Parameter entfernen.

Reply:
Ja da hast du recht, ursprünglich wollte ich glaube ich verschiedene Funktionen in der Klasse verwenden...
Hab die parameter entfernt und den Konstruktor analog zum "feedback_exporter" beibehalten.
133d532afd97506c96e5730ab0acca9161b6b022